cpp/ql/lib/semmle/code/cpp/exprs/Literal.qll

```diff
@@ -261,11 +289,38 @@ class ArrayOrVectorAggregateLiteral extends AggregateLiteral {
   Type getElementType() { none() }
 
   /**
+   * Gets an expression within the aggregate literal that is used to initialize
+   * element `elementIndex`, if present.
+   *
+   * This predicate may have multiple results since an element can be initialized
+   * multiple times in the same initializer.
+   */
+  Expr getAnElementExpr(int elementIndex) { result = this.getElementExpr(elementIndex, _) }
+
+  /**
+   * DEPRECATED: Use `getAnElementExpr` instead.
+   *
    * Gets the expression within the aggregate literal that is used to initialize
    * element `elementIndex`, if present.
+   *
+   * This predicate may have multiple results since an element can be initialized
+   * multiple times in the same initializer.
+   */
+  Expr getElementExpr(int elementIndex) { result = this.getElementExpr(elementIndex, _) }
+
+  /**
+   * Gets the expression within the aggregate literal that is used to initialize
+   * element `elementIndex`, if present. The expression is the `position`'th entry
+   * in the aggregate literal.
+   *
+   * For example, if `a` represents the initialization literal `{[0] = 1234, [0] = 5678}` in
+   * ```cpp
+   * int x[1] = {[0] = 1234, [0] = 5678};
+   * ```
+   * then `a.getElementExpr(0, 0)` gives `1234`, and `a.getElementExpr(0, 1)` gives `5678`.
```

Comment:
I think this example is somewhat confusing, as we'll also have:
```cpp
int x[2] = {[0] = 1234, [1] = 5678};
```
with `a.getElementExpr(0, 0) = 1234` and `a.getElementExpr(1, 1) = 5678`.

Reply:
Hm, true. Should we just replace my example with yours (and let the case of repeated initialization be implicit)?

Reply:
Or maybe a combination like:
```
int x[2] = {[0] = 123, [1] = 456, [0] = 789 };
```
?

Reply:
Fixed in 6c26ad7c2dc2282685b927be8343930712b695da.